c++/include/orc/Vector.hh

```diff
@@ -140,6 +140,11 @@ namespace orc {
      */
     virtual void resize(uint64_t capacity);
 
+    /**
+     * Heap memory used by the batch.
+     */
+    virtual int64_t memoryUse();
```

Comment:
I think the use of -1 as a special value is problematic. I think you should have it return the current usage for all types and add a method hasVariableLength() that returns if it contains a list or map.
Please use uint64_t.
Please rename to getMemoryUsage().